dustin/async_file_reader.js

```diff
@@ -0,0 +1,35 @@
+
+const EventEmitter = require('events').EventEmitter;
+const fs = require('fs');
+const ee = new EventEmitter();
+files = ['./one.txt', './two.txt', './three.txt'];
+output = {};
+logger = ""
+
+ee.on('check-and-log', (data) => {
+  if (Object.keys(output).length == files.length) {
+    for (i = 1; i < Object.keys(output).length + 1; i++) {
+      logger = logger + output[i].slice(0, 8).toString('hex') + "\n"
+
+    }
+console.log(logger);
+  }
+})
+
+readfile = function (file, order) {
+  fs.readFile(file, (err, data) => {
+    if (err) return console.log(err);
+    output[order] = data;
```

Comment:
If all you're using for the key in output is a number you can use an array here. If the variable output was an array it would have pretty much been the same and you wouldn't need to do the Object.keys bit for manipulation.